network/src/processor/naive_transition_frontier.rs

```diff
@@ -38,14 +39,19 @@ impl TransitionFrontier for NaiveTransitionFrontier {
     }
 
     async fn add_block(&mut self, block: Self::Block) -> anyhow::Result<()> {
-        if self.best_chain.length() < 1 {
-            self.best_chain.push(block.protocol_state)?;
+        let mut ctx = mina_signer::create_legacy::<SignedCommandPayload>(NetworkId::MAINNET);
```

Comment:
This crate might be an appropriate place to make this a `static mut` somewhere. Not sure about this though

Reply:
To reuse ctx in a thread-safe way, I would use an object pool here